subprojects/core/src/main/java/org/gradle/util/NameValidator.java

```diff
@@ -33,9 +33,9 @@ private NameValidator() { }
      */
     public static void validate(String name) {
         if (StringUtils.containsAny(name, FORBIDDEN_CHARACTERS)) {
-            DeprecationLogger.nagUserOfDeprecated("The name '" + name + "' contains at least one of the following characters: " + Arrays.toString(FORBIDDEN_CHARACTERS) + ". This");
+            DeprecationLogger.nagUserOfDeprecatedThing("The name '" + name + "' contains at least one of the following characters: " + Arrays.toString(FORBIDDEN_CHARACTERS));
```

Comment:
:o: I think we should find a better word than "thing". How about "GradleDomainObject"? 

...I just saw that we already have been using "thing" before. ;-)